src/main/java/org/sagebionetworks/web/server/servlet/filter/CrawlFilter.java

```diff
@@ -445,6 +479,72 @@ private String getEntityHtml(String entityId)
     return html.toString();
   }
 
+  public String getDatasetScriptElement(EntityBundle bundle)
+    throws JSONObjectAdapterException, RestServiceException {
+    StringBuilder html = new StringBuilder();
+    if (bundle.getEntity() instanceof Dataset) {
+      Dataset ds = (Dataset) bundle.getEntity();
+      html.append("<script type=\"application/ld+json\">");
+      JSONObjectAdapter json = new JSONObjectAdapterImpl();
+      json.put("@context", "http://schema.org/");
+      json.put("@type", "Dataset");
+      json.put("name", ds.getName());
+      json.put("description", ds.getDescription());
+      json.put(
+        "url",
+        "https://www.synapse.org/#!Synapse:" +
+        ds.getId() +
+        "." +
+        ds.getVersionNumber()
+      );
+      json.put("version", ds.getVersionNumber());
+
+      // add annotations
+      JSONArrayAdapter array = new JSONArrayAdapterImpl();
+      Map<String, AnnotationsValue> annotations = bundle
+        .getAnnotations()
+        .getAnnotations();
+      Set<String> annotationKeys = annotations.keySet();
+      int index = 0;
+      for (String key : annotationKeys) {
+        List<String> keyValuePairList = new ArrayList<String>();
+        keyValuePairList.add(key);
+        List<String> values = annotations.get(key).getValue();
+        for (String value : values) {
+          keyValuePairList.add(value);
+        }
+        array.put(index++, String.join(", ", keyValuePairList));
+      }
+      json.put("keywords", array);
+
+      // include identifier if there is a DOI association?
+
+      JSONObjectAdapter object = new JSONObjectAdapterImpl();
+      object.put("@type", "DataCatalog");
+      object.put("name", "Synapse");
+      object.put("url", "https://www.synapse.org");
```

Comment:
Would be cool if we had a specific instance of this Synapse DataCatalog object in the semantic web, and then maybe Google has a feature where you could reverse index and find all public datasets in Synapse!